test_framework/tacky/common.py

```diff
@@ -220,3 +220,17 @@ def is_zero_instr(i: asm.AsmItem) -> bool:
         and i.opcode == Opcode.XOR
         and i.operands[0] == i.operands[1]
     )
+
+def check_constant_folds(parsed_asm, program_source_file):
```

Comment:
This new function needs a doc string and mypy type annotations. I'd also give it a more general name, like `check_instructions`, since there's nothing constant folding-specific about this function except the error message (the constant folding-specific logic is all in the `ok` function defined in the callers).

Reply:
Oh, I see. That makes sense. I will make those changes.